testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeCheckConstraintCompatibility.java

```diff
@@ -104,6 +104,39 @@ public static Object[][] checkConstraints()
         };
     }
 
+    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_OSS, DELTA_LAKE_EXCLUDE_73, PROFILE_SPECIFIC_TESTS})
+    @Flaky(issue = DATABRICKS_COMMUNICATION_FAILURE_ISSUE, match = DATABRICKS_COMMUNICATION_FAILURE_MATCH)
```

Comment:
Why are we introducing a flaky test?

Reply:
All Databricks product tests need this annotation because they throw 503 error occasionally. We decided to retry the test method instead of adding retry logic to `onDelta` method because DML could succeed in Databricks side even if they return the error.

Reply:
That's a big code smell IMO. The purpose of annotating test as flaky is to avoid spurious test failures until someone goes and improves the test. It should not be used as a mechanism for automatically retrying tests.

If the underlying infrastructure is unreliable, then the test needs to account for that by having a retry mechanism *in the test* itself.

Reply:
It's pre-existing issue and I don't think starting the discussion in this PR is a good idea. Let us discuss the topic internally.

Reply:
In that case, I would argue we shouldn't add the test until we can test it properly, or unless we're planning to immediately follow up with a fix. We should avoid accumulating technical debt.

Reply:
Removed `@Flaky` annotatin, `DELTA_LAKE_DATABRICKS` and `DELTA_LAKE_EXCLUDE_73` groups.